resources/lib/kodi/library.py

```diff
@@ -452,19 +450,24 @@ def export_new_episodes(videoid, scan=True):
     :param scan: Whether or not to scan the library after exporting, useful for a single show
     :return: None
     """
-
-    for task in _create_new_episodes_tasks(videoid):
-        try:
-            export_item(task, library_path())
-        except Exception:
-            import traceback
-            common.error(traceback.format_exc())
-            common.error('{} of {} failed'
-                         .format(export_item.__name__, task['title']))
-
-    if scan:
-        common.debug('Notify service to update the library')
-        common.send_signal(common.Signals.LIBRARY_UPDATE_REQUESTED)
+    if videoid.mediatype == common.VideoId.SHOW:
+        # First let's fetch metadata of the show from api
+        metadata = api.metadata(videoid, True)
+        if metadata and 'seasons' in metadata[0]:
+            for task in _create_new_episodes_tasks(videoid, metadata):
+                try:
+                    export_item(task, library_path())
+                except Exception:
+                    import traceback
+                    common.error(traceback.format_exc())
+                    common.error('{} of {} failed'
+                                 .format(export_item.__name__, task['title']))
+
+            if scan:
+                common.debug('Notify service to update the library')
+                common.send_signal(common.Signals.LIBRARY_UPDATE_REQUESTED)
+    else:
```

Comment:
wrong indent

Reply:
The else is for `if videoid.mediatype == common.VideoId.SHOW:` `else:`

If not a tvshow, debug message. I plan to add an 'export new episodes' context menu on tvshow, it's to ensure it won't be started for anything else (ie with a xbmc.executebuildin from ananother addon with wrong parameters)